Makefile

```
@@ -1,61 +1,31 @@
-CARGO = cargo --color always
 RUSTUP = rustup
 TOOLCHAIN = $(shell cat rust-toolchain)
-TOOLCHAIN_LOC = ""
-
-# Platform specific magic
-LIBUSB_LOC = ""
-LIBUSB_INSTALL = ""
-UNAME_S := $(shell uname -s)
-UNAME_M := $(shell uname -m)
-ifeq ($(UNAME_S),Linux)
-	LIBUSB_LOC = /usr/include/libusb-1.0
-	LIBUSB_INSTALL = sudo apt-get install --assume-yes libusb-1.0-0-dev
-	TOOLCHAIN_LOC = $(RUSTUP_DIR)/toolchains/$(TOOLCHAIN)-$(UNAME_M)-unknown-linux-gnu
-endif
-ifeq ($(UNAME_S),Darwin)
-	LIBUSB_LOC = /usr/local/lib/libusb-1.0.dylib
-	LIBUSB_INSTALL = brew install libusb
-	TOOLCHAIN_LOC = $(RUSTUP_DIR)/toolchains/$(TOOLCHAIN)-$(UNAME_M)-apple-darwin
-endif
-
-CARGO_DIR = $(HOME)/.cargo
-RUSTUP_DIR = $(HOME)/.rustup
-
-CLIPPY_LOC = $(TOOLCHAIN_LOC)/bin/cargo-clippy
-FMT_LOC = $(TOOLCHAIN_LOC)/bin/cargo-fmt
-TOMLFMT_LOC = $(CARGO_DIR)/bin/cargo-tomlfmt
+CARGO = $(RUSTUP) run --install $(TOOLCHAIN) cargo --color always
 
-build: build_debug
-
-## Build Environment
-# `test` is used to only run if the target does not exist
-# Instead of running if the pre-requisite is older than the target
+CLIPPY_LOC = $(shell which cargo-clippy)
```

This would make it more consistent? :)
```suggestion
CLIPPY = $(shell which cargo-clippy)
```